fraction.rb

```diff
@@ -1,17 +1,64 @@
 class Fraction
   def initialize(numerator, denominator)
-    if !(numerator.is_a?(Integer) && denominator.is_a?(Integer))
-      # TODO: create a proper exception for that issue
-      # TODO: handle case when one or both parameters are not Integer but could be easily transformed into it
-      # by using #to_i
-      raise 'At least some arguments are incorrect: ' + numerator.inspect + ', ' + denominator.inspect
+    if ! numerator.is_a?(Integer) && denominator.is_a?(Integer)
```

Comment:
You should manage these validations separately: one check for `numerator` and other for `denominator`.
And makes sense to create new kind of exception for it.